ravenframework/SupervisedLearning/DMD/EDMD.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on July 21, 2024
+
+  @author: Andrea Alfonsi
+  Kernalized Dynamic Mode Decomposition
+
+"""
+#Internal Modules (Lazy Importer)--------------------------------------------------------------------
+from ...utils.importerUtils import importModuleLazy
+#Internal Modules (Lazy Importer) End----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+np = importModuleLazy("numpy")
+pydmd = importModuleLazy("pydmd")
+ezyrb = importModuleLazy("ezyrb")
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from ...SupervisedLearning.DMD import DMDBase
+from ...utils import InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class EDMD(DMDBase):
+  """
+    Kernelized Extended Dynamic Mode Decomposition (Parametric)
+  """
+  info = {'problemtype':'regression', 'normalize':False}
+
+  def __init__(self):
+    """
+      Constructor that will appropriately initialize a supervised learning object
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+
+    # local model
+    self._dmdBase = None # {} # CDMD
```

Comment:
default is None, you may not need to redefine it here.

Reply:
you are right. Removed everywhere.